src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -416,36 +461,42 @@ private AIState clearStep()
 
     private AIState requestMaterials()
     {
+        if(job.getStructure() == null)
+        {
+            //fix for bad structures
+            job.complete();
+        }
+
         //todo as soon as material handling has been implemented this should be set to work!
         //We need to deal with materials
         if (!Configurations.builderInfiniteResources)
         {
             //TODO thread this
-            while (job.getSchematic().findNextBlock())
+            while (job.getStructure().findNextBlock())
             {
-                if (job.getSchematic().doesSchematicBlockEqualWorldBlock())
+                if (job.getStructure().doesStructureBlockEqualWorldBlock())
                 {
                     continue;
                 }
 
-                @Nullable Block block = job.getSchematic().getBlock();
+                @Nullable Block block = job.getStructure().getBlock();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'block' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)